src/haz3lweb/ModelSettings.re

```
@@ -1,5 +1,20 @@
 open Sexplib.Std;
 
+[@deriving (show({with_path: false}), sexp, yojson)]
+type mode =
+  | DebugLoad
+  | Scratch
+  | Examples
+  | Exercise;
+
+let mode_of_string = (s: string): mode =>
+  switch (s) {
+  | "Scratch" => Scratch
+  | "Examples" => Examples
+  | "Exercise" => Exercise
+  | _ => Scratch
```

failwith rather than defaulting here to catch potential future bugs